atoma-service/src/middleware.rs

```diff
@@ -334,8 +335,31 @@ pub async fn verify_stack_permissions(
             StatusCode::UNAUTHORIZED
         })?;
     if available_stack.is_none() {
-        error!("No available stack with enough compute units");
-        return Err(StatusCode::UNAUTHORIZED);
+        let tx_digest_str = req_parts
+            .headers
+            .get("X-Tx-Digest")
+            .ok_or_else(|| {
+                error!("Stack not found, tx digest header expected but not found");
+                StatusCode::BAD_REQUEST
+            })?
+            .to_str()
+            .map_err(|_| {
+                error!("Tx digest cannot be converted to a string");
+                StatusCode::BAD_REQUEST
+            })?;
+        let tx_digest = TransactionDigest::from_str(tx_digest_str).unwrap();
+        let (tx_stack_small_id, compute_units) =
+            utils::request_blockchain_for_stack(&state, tx_digest, total_num_compute_units).await?;
+
+        // NOTE: We need to check that the stack small id matches the one in the request
+        // otherwise, the user is requesting for a different stack, which is invalid. We
+        // must also check that the compute units are enough for processing the request.
+        if stack_small_id != tx_stack_small_id as i64
+            || compute_units > total_num_compute_units as u64
```

Comment:
at this stage, the compute units check should already be done.
it should be done at the `insert_new_stack`, and there should be some logic on conflict.